### ISSUE TITLE
[Accessibility]: Search boxes in sidebar sections should have different labels

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43146]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

The labels on search boxes in search sidebar sections are all labelled the same. This makes it impossible for a screen reader user to distinguish them when navigating by form elements.

<img width="273" alt="image" src="https://user-images.githubusercontent.com/206864/196561075-b5b4283f-e869-4fbe-a846-b811d87b420a.png">

### Expected behavior

The search boxes in the sidebar have labels based on which section they're in.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard
